gencert-private.sh

```diff
@@ -5,11 +5,23 @@ key="key.pem"
 
 # check organizationIdentifier valid value (VATID-XXX... or CF:IT-XXX..)
 
-if echo "$ORGANIZATION" | grep -qe 'CF:IT-*' || echo "$ORGANIZATION" | grep -qe 'VATID-*'; then  
-    echo $ORGANIZATION valid value
+if [ $(echo ${ORGANIZATION_IDENTIFIER} | grep -c -P "^(CF:IT-[\d\w]{16})|^(VAT[A-Z]{2}-\d{11})") -ge 1 ]; then  
+    echo $ORGANIZATION_IDENTIFIER valid value
 else
-    echo $ORGANIZATION not valid value
-    exit
+    echo $ORGANIZATION_IDENTIFIER not valid value for organizationIdentifier
+    exit 1
+fi
```

Comment:
Having multiple occurrences (`-ge 1`) of the regexp  does not represent a good result. As for `KEY_LEN` and `MD_ALG`, we're ok if we obtain only one occurrence. Furthermore, we're only interested in terminating the scrip if the validation fails. Therefore, I would update the check as in the following

```
if [ $(something) -ne 1 ]; then
    echo "[E] Error message here"
    exit 1
fi
```